Zend/Optimizer/sccp.c

```diff
@@ -800,57 +800,115 @@ static inline zend_result ct_eval_array_key_exists(zval *result, zval *op1, zval
-		|| zend_string_equals_literal(name, "base64_decode")
-		|| zend_string_equals_literal(name, "base64_encode")
+	 * or warn on invalid arguments, as we detect this and will discard the evaluation result.
+	 *
+	 * In PHP 8.2, many functions stopped depending on locales due to https://wiki.php.net/rfc/strtolower-ascii
+	 *
+	 * Break up a really long if-else chain that the compiler can't optimize by switching on the length for the following reasons:
+	 * 1. Improve worst-case performance.
+	 * 2. Make it less of a concern whether a function is used often enough to include in this list. */
+	switch (ZSTR_LEN(name)) {
+		CAN_CT_EVAL_FUNC_CALL_CHECK_CASE(3, false
+			|| zend_string_equals_literal(name, "pow")
+		);
+		CAN_CT_EVAL_FUNC_CALL_CHECK_CASE(4, false
+			|| zend_string_equals_literal(name, "trim")
+		);
+		CAN_CT_EVAL_FUNC_CALL_CHECK_CASE(5, false
+			|| zend_string_equals_literal(name, "crc32")
+			|| zend_string_equals_literal(name, "ltrim")
+			|| zend_string_equals_literal(name, "rtrim")
+		);
+		CAN_CT_EVAL_FUNC_CALL_CHECK_CASE(6, false
+			|| zend_string_equals_literal(name, "strcmp")
+			|| zend_string_equals_literal(name, "strpos")
+			|| zend_string_equals_literal(name, "strstr")
+			|| zend_string_equals_literal(name, "substr")
+		);
+		CAN_CT_EVAL_FUNC_CALL_CHECK_CASE(7, false
 #ifndef ZEND_WIN32
-		/* On Windows this function may be code page dependent. */
-		|| zend_string_equals_literal(name, "dirname")
+			/* On Windows this function may be code page dependent. */
+			|| zend_string_equals_literal(name, "dirname")
 #endif
-		|| zend_string_equals_literal(name, "explode")
-		|| zend_string_equals_literal(name, "imagetypes")
-		|| zend_string_equals_literal(name, "in_array")
-		|| zend_string_equals_literal(name, "implode")
-		|| zend_string_equals_literal(name, "ltrim")
-		|| zend_string_equals_literal(name, "php_sapi_name")
-		|| zend_string_equals_literal(name, "php_uname")
-		|| zend_string_equals_literal(name, "phpversion")
-		|| zend_string_equals_literal(name, "pow")
-		|| zend_string_equals_literal(name, "preg_quote")
-		|| zend_string_equals_literal(name, "rawurldecode")
-		|| zend_string_equals_literal(name, "rawurlencode")
-		|| zend_string_equals_literal(name, "rtrim")
-		|| zend_string_equals_literal(name, "serialize")
-		|| zend_string_equals_literal(name, "str_contains")
-		|| zend_string_equals_literal(name, "str_ends_with")
-		|| zend_string_equals_literal(name, "str_replace")
-		|| zend_string_equals_literal(name, "str_split")
-		|| zend_string_equals_literal(name, "str_starts_with")
-		|| zend_string_equals_literal(name, "strpos")
-		|| zend_string_equals_literal(name, "strstr")
-		|| zend_string_equals_literal(name, "substr")
-		|| zend_string_equals_literal(name, "trim")
-		|| zend_string_equals_literal(name, "urldecode")
-		|| zend_string_equals_literal(name, "urlencode")
-		|| zend_string_equals_literal(name, "version_compare")
-	) {
-		return true;
+			|| zend_string_equals_literal(name, "explode")
```

Comment:
`bin2hex` `hex2bin` `strlen` `mb_strlen` would be good here as well, maybe also `preg_match` `preg_replace`

Reply:
See below comment